core/qt-ble.cpp

```diff
@@ -65,21 +68,24 @@ void BLEObject::characteristcStateChanged(const QLowEnergyCharacteristic &c, con
 {
 	if (IS_HW(device)) {
 		if (c.uuid() == hwAllCharacteristics[HW_OSTC_BLE_DATA_TX]) {
+			hw_credit--;
 			receivedPackets.append(value);
+			if (hw_credit == MINIMAL_HW_CREDIT)
+				setHwCredit(MAXIMAL_HW_CREDIT - hw_credit);
 		} else {
 			qDebug() << "ignore packet from" << c.uuid() << value.toHex();
 		}
 	} else {
 		receivedPackets.append(value);
 	}
-	//qDebug() << ".. incoming packet count" << receivedPackets.length();
 }
 
 void BLEObject::characteristicWritten(const QLowEnergyCharacteristic &c, const QByteArray &value)
 {
 	if (IS_HW(device)) {
 		if (c.uuid() == hwAllCharacteristics[HW_OSTC_BLE_CREDITS_RX]) {
-			qDebug() << "HW_OSTC_BLE_CREDITS_RX confirmed" << c.uuid() << value.toHex();
+			bool ok;
+			hw_credit += value.toHex().toInt(&ok, 16);
```

Comment:
To be honest ... I just wanted a simple way to convert a QByteArray with just one byte to a unsigned int (0 -255)... and all I tried returned something unexpected except this one. So a poor man's choice.